src/wallet/wallet.cpp

```diff
@@ -3166,15 +3166,17 @@ bool CWallet::CreateTransaction(const std::vector<CRecipient>& vecSend, CTransac
                         }
 
                         std::vector<CTxOut>::iterator position = txNew.vout.begin()+vChangePosInOut[assetChange.first];
-                        txNew.vout.insert(position, newTxOut);
-                        CPubKey blind_pub = GetBlindingPubKey(itScript->second.second);
                         if (blind_details) {
+                            CPubKey blind_pub = GetBlindingPubKey(itScript->second.second);
                             blind_details->o_pubkeys.insert(blind_details->o_pubkeys.begin() + vChangePosInOut[assetChange.first], blind_pub);
                             assert(blind_pub.IsFullyValid());
                             blind_details->num_to_blind++;
                             blind_details->change_to_blind++;
                             blind_details->only_change_pos = vChangePosInOut[assetChange.first];
+                            // Place the blinding pubkey here in case of fundraw calls
+                            newTxOut.nNonce.vchCommitment = std::vector<unsigned char>(blind_pub.begin(), blind_pub.end());
```

Comment:
I already commented in chat, but perhaps good to put it in public: how come `blindrawtransaction` doesn't blind this output without this nonce? `blindrawtransaction` is a wallet call, so I would suspect it to blind all outputs that belong to the wallet, by redoing the `GetBlindingPubKey` lookup.

It's good to do this here, it makes sense, but f.e. when doing a `createrawtransaction` with wallet outputs, this would still mean `blindrawtransaction` doesn't blind all outputs it could blind unless the nonce is provided.

Reply:
It seems like this diff would do that:
```diff
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index 23b11c69b..bba0d1999 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -5348,10 +5348,11 @@ void FillBlinds(CWallet* pwallet, CMutableTransaction& tx, std::vector<uint256>&
         if (out.nValue.IsExplicit()) {
             CPubKey pubkey(out.nNonce.vchCommitment);
             if (!pubkey.IsFullyValid()) {
-                output_pubkeys.push_back(CPubKey());
-            } else {
-                output_pubkeys.push_back(pubkey);
+                // Try find the blinding pubkey in the wallet.
+                // (returns CPubKey() when not found)
+                pubkey = pwallet->GetBlindingPubKey(out.scriptPubKey);
             }
+            output_pubkeys.push_back(pubkey);
             output_value_blinds.push_back(uint256());
             output_asset_blinds.push_back(uint256());
         } else if (out.nValue.IsCommitment()) {
```